docs/developer-guide.md

```diff
@@ -43,6 +43,20 @@ You can undeploy Katib v1beta1 manifests from a Kubernetes cluster as follows:
 make undeploy
 ```
 
+## Technical and style guide
+
+The following guidelines apply primarily to Katib, 
+but other projects like training-operator might also adhere to them. 
+
+When coding:
+
+- Follow [effective go](https://go.dev/doc/effective_go) guidelines.
+
+Testing:
+
+- Use `cmp.Diff` instead of `reflect.Equal`, to provide useful comparisons.
```

Comment:
Would be nice to provide link to `cmp.Diff`.

Reply:
That makes sense.

Reply:
Done.